rpc/service/src/service.rs

```diff
@@ -609,13 +610,31 @@ NOTE: This error usually indicates an RPC conversion error between the node and
         request: GetVirtualChainFromBlockRequest,
     ) -> RpcResult<GetVirtualChainFromBlockResponse> {
         let session = self.consensus_manager.consensus().session().await;
-        let virtual_chain = session.async_get_virtual_chain_from_block(request.start_hash).await?;
+
+        // If start_hash is empty - use source instead.
+        let start_hash = if request.start_hash.is_none() {
+            // we default to the pruning point if the start hash is not provided
+            session.async_pruning_point().await
+        } else {
+            request.start_hash
+        };
+
+        // limit is set to 10 times the mergeset_size_limit.
+        // this means limit is 2480 on 10 bps, and 1800 on mainnet.
+        // this bounds by number of merged blocks, if include_accepted_transactions = true
+        // else it returns the limit amount on pure chain blocks.
+        let limit = (self.config.mergeset_size_limit * 10) as usize;
+        let mut virtual_chain_batch = session.async_get_virtual_chain_from_block(start_hash, limit).await?;
```

Comment:
No, removed has no associated accepted txs info (it relies on the client to track what was previously accepted by the removed chain blocks), and we don't limit its count (bcs it can't be longer than a reorg which is naturally bounded)